vsphere/resource_vsphere_vapp_container.go

```diff
@@ -0,0 +1,369 @@
+	if err != nil {
+		return err
+	}
+	rpSpec := expandVAppContainerConfigSpec(d)
+	vcSpec := &types.VAppConfigSpec{}
+	var f *object.Folder
+	if pf, ok := d.GetOk("parent_folder_id"); ok {
+		f, err = folder.FromID(client, pf.(string))
+		if err != nil {
+			return err
+		}
+	} else {
+		p := strings.Split(prp.InventoryPath, "/")
+		if len(p) > 2 {
+			return fmt.Errorf("unable to locate datacenter name from parent resource pool")
+		}
+		dc, err := getDatacenter(client, p[1])
+		if err != nil {
+			return err
+		}
+		f, err = folder.FromPath(client, "", folder.VSphereFolderTypeVM, dc)
+		if err != nil {
+			return err
+		}
+	}
+	vc, err := vappcontainer.Create(prp, d.Get("name").(string), rpSpec, vcSpec, f)
+	if err != nil {
+		return err
+	}
+	if err = resourceVSphereVAppContainerApplyTags(d, meta, vc); err != nil {
+		return err
+	}
+	d.SetId(vc.Reference().Value)
+	log.Printf("[DEBUG] %s: Create finished successfully", resourceVSphereVAppContainerIDString(d))
+	return resourceVSphereVAppContainerRead(d, meta)
+}
+
+func resourceVSphereVAppContainerRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning read", resourceVSphereVAppContainerIDString(d))
+	client, err := resourceVSphereVAppContainerClient(meta)
+	if err != nil {
+		return err
+	}
+	vc, err := vappcontainer.FromID(client, d.Id())
+	if err != nil {
+		if viapi.IsManagedObjectNotFoundError(err) {
+			log.Printf("[DEBUG] %s: Resource has been deleted", resourceVSphereVAppContainerIDString(d))
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	if err = resourceVSphereVAppContainerReadTags(d, meta, vc); err != nil {
+		return err
+	}
+	if err = d.Set("name", vc.Name()); err != nil {
+		return err
+	}
+	vcProps, err := vappcontainer.Properties(vc)
+	if err != nil {
+		return err
+	}
+	if err = d.Set("parent_resource_pool_id", vcProps.Parent.Value); err != nil {
+		return err
+	}
+	if err = d.Set("parent_folder_id", vcProps.ParentFolder.Value); err != nil {
+		return err
+	}
+	if err = flattenVAppContainerConfigSpec(d, vcProps.Config); err != nil {
+		return err
+	}
+	log.Printf("[DEBUG] %s: Read finished successfully", resourceVSphereVAppContainerIDString(d))
+	return nil
+}
+
+func resourceVSphereVAppContainerUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning update", resourceVSphereVAppContainerIDString(d))
+	client, err := resourceVSphereVAppContainerClient(meta)
+	if err != nil {
+		return err
+	}
+	vc, err := vappcontainer.FromID(client, d.Id())
+	if err != nil {
+		return err
+	}
+	if err = resourceVSphereVAppContainerApplyTags(d, meta, vc); err != nil {
+		return err
+	}
+	op, np := d.GetChange("parent_resource_pool_id")
+	if op != np {
+		log.Printf("[DEBUG] %s: Parent resource pool has changed. Moving from %s, to %s", resourceVSphereVAppContainerIDString(d), op, np)
+		p, err := resourcepool.FromID(client, np.(string))
```

Comment:
This gives me a warning on `vetshadow` as a redundant err declaration and can probably be fixed to just be an assignment without declaration (`=` instead of `:=`).

Reply:
Just referencing https://github.com/terraform-providers/terraform-provider-vsphere/pull/566#discussion_r203573010 here